diff -ruN pdp-0.12.5.old/modules/generic/pdp_rawout.c pdp-0.12.5/modules/generic/pdp_rawout.c
--- pdp-0.12.5.old/modules/generic/pdp_rawout.c	2009-10-29 13:43:00.000000000 +0100
+++ pdp-0.12.5/modules/generic/pdp_rawout.c	2009-10-29 13:43:12.000000000 +0100
@@ -138,7 +138,7 @@
 
     /* open pipe */
     if (-1 == (pipefd = open(x->x_pipe->s_name, O_WRONLY|O_NONBLOCK|O_APPEND))){
-	if (-1 == (pipefd = open(x->x_pipe->s_name, O_WRONLY|O_CREAT))){
+	if (-1 == (pipefd = open(x->x_pipe->s_name, O_WRONLY|O_CREAT, S_IRUSR|S_IWUSR))){
 	    perror(me);
 	    goto exit;
 	}
